x-pack/solutions/security/plugins/security_solution/public/data_view_manager/containers/data_view_provider.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { FC, PropsWithChildren } from 'react';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { useKibana } from '../../common/lib/kibana';
+import { createDefaultDataView } from '../utils/create_default_data_view';
+
+const DefaultDataViewContext = createContext<DataView | undefined>(undefined);
+
+export const useDefaultDataView = () => {
+  const defaultDataView = useContext(DefaultDataViewContext);
+
+  if (!defaultDataView) {
+    throw new Error('Could not fetch default data view');
+  }
+
+  return defaultDataView;
+};
+
+export const DefaultDataViewProvider: FC<PropsWithChildren> = ({ children }) => {
+  const { dataViews, uiSettings, spaces, application, http } = useKibana().services;
+
+  const [defaultDataView, setDefaultDataView] = useState<DataView>();
+
+  useEffect(() => {
+    (async () => {
+      const { defaultDataView: dv } = await createDefaultDataView({
+        dataViewService: dataViews,
+        uiSettings,
+        spaces,
+        application,
+        http,
+      });
+
+      const dataView = await dataViews?.get(dv.id);
```

Comment:
Even though it's cached, we can just return the _actual_ dataView here https://github.com/elastic/kibana/blob/8d605fd48e0c6a4af482b23468da71c61b2be2ff/x-pack/solutions/security/plugins/security_solution/public/sourcerer/containers/create_sourcerer_data_view.ts#L89 for the time being?